unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDView.cs

```diff
@@ -25,7 +27,17 @@ public static ChatHUDView Create()
     public void Initialize(ChatHUDController controller, UnityAction<string> onSendMessage)
     {
         this.controller = controller;
-        inputField.onSubmit.AddListener(onSendMessage);
+        onSendMessageAction = onSendMessage;
+        inputField.onSubmit.AddListener(OnInputFieldSubmit);
+    }
+
+    private void OnInputFieldSubmit(string message)
+    {
+        // A TMP_InputField is automatically marked as 'wasCanceled' when the ESC key is pressed
+        if (inputField.wasCanceled)
```

Comment:
We should add a test to cover this behaviour

Reply:
Test added!